diff -c -r src-bak//io/writedataavs.c src//io/writedataavs.c
*** src-bak//io/writedataavs.c	2012-04-02 13:56:30.000000000 +0100
--- src//io/writedataavs.c	2012-04-02 13:36:42.000000000 +0100
***************
*** 102,108 ****
  
      for (int i=0; i<gdata->n_vert; i++) {
        ierr = PetscFPrintf(PETSC_COMM_WORLD,fd,
!         "%i %g %g %g\n",
          i+1,
          PetscRealPart(gdata->vertxyz[ND*i+0]),
          PetscRealPart(gdata->vertxyz[ND*i+1]),
--- 102,108 ----
  
      for (int i=0; i<gdata->n_vert; i++) {
        ierr = PetscFPrintf(PETSC_COMM_WORLD,fd,
!         "%i %0.15e %0.15e %0.15e\n",
          i+1,
          PetscRealPart(gdata->vertxyz[ND*i+0]),
          PetscRealPart(gdata->vertxyz[ND*i+1]),
***************
*** 203,213 ****
      snprintf(
        wbufe,
        nmax-(wbufe-wbufs),
!       "%i %g %g %g %g %g %g %g %g %g "
  #ifdef EXCH
!       "%g %g %g "
  #endif
!       "%g %g %g\n",
        gdata->vertl2g[i]+1,
        ta_M[ND*i+0],
        ta_M[ND*i+1],
--- 203,213 ----
      snprintf(
        wbufe,
        nmax-(wbufe-wbufs),
!       "%i %0.15e %0.15e %0.15e %0.15e %0.15e %0.15e %0.15e %0.15e %0.15e "
  #ifdef EXCH
!       "%0.15e %0.15e %0.15e "
  #endif
!       "%0.15e %0.15e %0.15e\n",
        gdata->vertl2g[i]+1,
        ta_M[ND*i+0],
        ta_M[ND*i+1],
