diff -rupN nmag-0.2.1/hdf5/tools/lib/h5diff.c nmag-0.2.1-new/hdf5/tools/lib/h5diff.c
--- nmag-0.2.1/hdf5/tools/lib/h5diff.c	2014-05-02 13:30:42.439806290 +0100
+++ nmag-0.2.1-new/hdf5/tools/lib/h5diff.c	2014-05-02 13:33:50.979809830 +0100
@@ -635,7 +635,7 @@ hsize_t h5diff(const char *fname1,
     char         filenames[2][MAX_FILENAME];
     hsize_t      nfound = 0;
     int i;
-    //int i1, i2;
+    /*int i1, i2;*/
     int l_ret;
     const char * obj1fullname = NULL;
     const char * obj2fullname = NULL;
